.github/workflows/build_and_test.yml

```diff
@@ -175,13 +215,13 @@ jobs:
         # Ensure user have access to network devices instead of giving super-user right
         run: |
           sudo chmod a+rw /dev/bpf*
-          cmake -DLIGHT_PCAPNG_ZSTD=${{ matrix.config-zstd }} -DCMAKE_OSX_ARCHITECTURES=${{ matrix.arch }} -S . -B $BUILD_DIR
+          cmake -DLIGHT_PCAPNG_ZSTD=${{ matrix.config-zstd }} -DCMAKE_OSX_ARCHITECTURES=${{ matrix.arch }} -DBUILD_SHARED_LIBS=ON -DPCAPPP_BUILD_COVERAGE=ON -S . -B $BUILD_DIR
```

Comment:
why is it needed to build share libs for coverage to work? 🤔

Reply:
It is because of out of build of tutorials (@clementperon mentioned [here](https://github.com/seladb/PcapPlusPlus/pull/1027#discussion_r1087214317)) If it is static linked there is a need of coverage flags inside examples since Tutorials build out of tree

Reply:
I read the thread but I'm not sure I understand. Can you please elaborate?

Reply:
@seladb Sorry for late response. I'm busy nowadays. It is because of linker step. While the `Pcap++` libraries are being built because of the coverage flags compiler and linker adds coverage related functions from `llvm` or `gcov` libraries. Since the tutorials stage is after the library builds the out of tree build of tutorials uses the already compiled `Pcap++` binaries. So, during the linker stage the `llvm`/`gcov` functions should be linked (because `Pcap++` binaries require them) but since there is no information about it in CMakeLists of tutorials about coverage flags, linker does not have any idea about where the functions are located, and it raises an error of undefined reference.

So basically, there is a two option, adding coverage flags to tutorials `CMakeLists` and using a boolean to add coverage linker flags or compiling shared libraries. Since shared libraries already linked there is no need to additional change in CMake

Reply:
Last option is to add a find_dependency like we do for DPDk and PF_RING

Reply:
We're building the tutorials on all platforms, it only bothers `llvm/gcov`?

Reply:
Actually, it affects all platforms (except VS, it is a completely different story, OpenCppCoverage is non-intrusive but I don't have any deep knowledge how it works). Since compiler wraps all `Pcap++` function calls with coverage related functions internally during to build  (invisible to us), to generate coverage report on both build-time and runtime (`gcda`, `gcno` files) there is a dependency to coverage libraries for all systems _if the coverage build option is enabled for library build_, otherwise not. I mean, the problem is not related with tutorials itself. It is because how the library was built. If the coverage is enabled `libPcap++` etc. requires these invisible functions and if you want to use them you need to link `llvm/gcov` libraries too.

Reply:
got it, I just noticed you also changed the linux job to also build shared libs.

IMHO we shouldn't change our CI to build shared libs just because of coverage. So I guess the only option is to add the coverage flags to the tutorials `CMakeLists`?

@clementperon please let us know what you think

Reply:
But on the other hand, using shared libraries because of coverage also means to checking of shared lib builds 😁

Reply:
Yes, that's a good point 😄 
Maybe we should have another job to test shared libraries. We don't have to run it on all platforms, maybe pick just one Linux option (Ubuntu?). But that's probably a separate topic to discuss (and a separate PR).

@egecetin do you think you can fix the tutorials `CMakeLists` to work with the coverage flags?
Or maybe @clementperon has a simpler solution to make it work?

Reply:
Done. I worked on find_dependency option